tasks/haskell-stack.yml

```diff
@@ -1,43 +1,25 @@
 ---
-- name: Add FPCO PGP public key (Debian)
+- name: Add FPCO's PGP public key
   apt_key:
-    url: https://s3.amazonaws.com/download.fpcomplete.com/debian/fpco.key
+    url: 'https://s3.amazonaws.com/download.fpcomplete.com/{{ ansible_distribution | lower }}/fpco.key'
     state: present
-  when: ansible_distribution == "Debian"
-
-- name: Add FPCO PGP public key (Ubuntu)
-  apt_key:
-    url: https://s3.amazonaws.com/download.fpcomplete.com/ubuntu/fpco.key
-    state: present
-  when: ansible_distribution == "Ubuntu"
 
-- name: Add FPCO deb repository (Jessie)
+# There is no specific repo for Stretch
+- name: Add FPCO's deb repository (Debian)
   apt_repository:
     repo: 'deb http://download.fpcomplete.com/debian/jessie stable main'
     state: present
-  when: ansible_distribution_release == "jessie"
-
-- name: Add FPCO deb repository (Stretch)
-  apt_repository:
-    repo: 'deb http://download.fpcomplete.com/debian/jessie stable main'
-    state: present
-  when: ansible_distribution_release == "stretch"
-
-- name: Add FPCO deb repository (Trusty)
-  apt_repository:
-    repo: 'deb http://download.fpcomplete.com/ubuntu trusty main'
-    state: present
-  when: ansible_distribution_release == "trusty"
+  when: ansible_distribution == "Debian"
 
-- name: Add FPCO deb repository (Xenial)
+- name: Add FPCO's deb repository (Ubuntu)
```

Comment:
Could simplify tasks to use debian and for ubuntu something like:
```
- name: Add FPCO deb repository (Debian)		
  apt_repository:		
   repo: 'deb http://download.fpcomplete.com/{{ ansible_distribution | lower }}/{{ ansible_distribution_release | lower }} stable main'		
   state: present		
when: ansible_distribution == "debian"

- name: Add FPCO deb repository (Debian)		
  apt_repository:		
   repo: 'deb http://download.fpcomplete.com/{{ ansible_distribution | lower }} {{ ansible_distribution_release | lower }}  main'		
   state: present		
when: ansible_distribution == "Ubuntu"
```
 So we can reduce the duplicate task.
No sure if worth dive in subfolders like
```
tasks/
        stack/debian.yml
        stack/ubuntu.yml
        stack.yml
        main.yml
        etc ..
```
Including with a when in stack.yml